File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int l=0,h=nums.size()-1,m;
        vector<int> ans;
        int one=-1,two=-1;
        while(l<=h){
            m=l+(h-l)/2;
            if(nums[m]==target)
            {
                two=m;
                l=m+1;
            }
            else if(nums[m]>target) h=m-1;
            else l=m+1;
        }
        l=0,h=nums.size()-1;
        while(l<=h){
            m=l+(h-l)/2;
            if(nums[m]==target)
            {
                one=m;
                h=m-1;
            }
            else if(nums[m]>target) h=m-1;
            else l=m+1;
        }
        ans.push_back(one);
        ans.push_back(two);
        return ans;
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] ans=new int[2];
        int start = search(nums,target,true);
        int end = search(nums,target,false);
        ans[0]=start;
        ans[1]=end;
        return ans;
        }
    
    int search(int[] nums, int target, boolean searchleft){
        int ans=-1;
        int start = 0;
            int end = nums.length-1;
            int mid = start+(end-start)/2;
            while(start<=end){
                mid = start+(end-start)/2;
                if(target<nums[mid]){
                    end = mid-1;
                }
                else if(target>nums[mid]){
                    start = mid+1;
                }
                else{
                    ans=mid;
                    if(searchleft){
                        end=mid-1;
                    }
                    else{
                        start = mid+1;
                    }
                }
            }
        return ans;
    }
}